app/src/main/java/com/concordium/wallet/ui/common/BackendErrorHandler.kt

```diff
@@ -44,6 +44,9 @@ object BackendErrorHandler {
             0 -> {
                 R.string.app_error_backend_internal_server
             }
+            1->{
+                R.string.delegation_register_delegation_amount_insufficient_error
```

Comment:
There can be multiple reasons for the error code being `1`. You can see some of them here https://github.com/Concordium/concordium-wallet-proxy/blob/e8d3bcb4ecf8ce9ea2382d55edc3b3494a2de43d/src/Internationalization/En.hs#L232. So we cannot assume that this error type is always because of insufficient funds. In the case of insufficient funds the error message will be `Transaction rejected by the node` which is not helpful to the user.

The issue should instead be solved by having proper validation on the delegation update page where the amount is chosen. This is also how it already works if I try to increase my amount to something higher when I do not have the funds for it. In the same way, it should show insufficient funds, if I don't have enough funds to cover the cost of the transaction that reduces the delegated amount.

Reply:
@czerwix Can you please create a task for the followup?

Reply:
The changes made do not correctly check for insufficient funds when updating the delegation amount (no new checks have been added or changed to fix the issue, so perhaps we have misunderstood each other).

Please ensure that it checks that the user has enough funds when updating the delegation amount.